Buddies.Web/src/pages/_app.tsx

```diff
@@ -18,12 +21,21 @@ const MyApp = ({ Component, pageProps }: AppProps) => {
     },
   }), [prefersDarkMode]);
 
+  const [loaded, setLoaded] = useState(false);
+
+  useEffect(() => {
+    fetchToken().catch((err) => err).finally(() => setLoaded(true));
+  }, []);
+
   return (
     <ThemeProvider theme={theme}>
       <CssBaseline enableColorScheme />
-      <Layout>
-        <Component {...pageProps} />
-      </Layout>
+      {loaded
+          && (
+          <Layout>
+            <Component {...pageProps} />
+          </Layout>
+          )}
```

Comment:
Ideally you would also implement protected routes here. If you don't have time to do it with this pr, that's OK, create a separate one. Just don't mix it in with some other feature.

Reply:
👌, we'll do it in a separate PR once we need the implementation.